aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
+
+	 dnu packages add newPackage.1.0.0.nupkg c:\packageStore  
+	 
+For additional details, run the help command::
+
+	dnu packages add --help
+
+Using commands
+--------------
+
+A command is a named execution of a .NET entry point with specific arguments. You can define commands in your *project.json* file:
+
+.. code-block:: json
+
+    {
+      "version": "1.0.0",
+      "webroot": "wwwroot",
+      "exclude": [
+        "wwwroot"
+      ],
+      "dependencies": {
+        "Kestrel": "1.0.0-beta4",
+        "Microsoft.AspNet.Diagnostics": "1.0.0-beta4",
+        "Microsoft.AspNet.Hosting": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-beta4",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-beta4"
+      },
+      "commands": {
+		"web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5000",
+		"kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5001",
+		"gen": "Microsoft.Framework.CodeGeneration",
+		"ef": "EntityFramework.Commands"
+      },
+      "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
+      }
+    }
+
+The commands are a set of arguments that will be passed to the DNX. The first part of this statement must be an assembly with a Main entry point that the DNX will try to execute. 
+	
+You can then use the DNU to install commands globally on a machine::
+
+    dnu commands install MyCommand
+	
+In the ``commands`` section shown above, the EF assembly doesn't require any extra arguments so it is just the name of the assembly.
+
+The assembly listed in the commands should be pulled in by a package that your application depends on.	
+
+For more information on using and creating commands see :doc:`/dnx/commands`.
+
+Running your app
+----------------
+
+By specifying a command as an argument that is passed to the DNX, you can run your app locally. For instance, you can use the ``kestrel`` command (specified in the ``commands`` section of your project's *project.json* file) to run a Web Application as follows::
+
+    dnx . kestrel
+
+To view the running web application, you can open a browser and navigate to the path specified in the *project.json* file::
+
+	http://localhost:5001
+	
+To run a Console Application using the Command Window from the project directory, you can use the following statement::
+
+	dnx . run
+	
+To run a Web API application, you can use the following command from the Command Window::
+
+	dnx . web
+	
+In the case of the *web* command, the ``Microsoft.AspNet.Hosting`` assembly has a ``Main`` entry point method that is called whenever you run the web command. The rest of the values in the *web* command are used by the hosting assembly to host your web application.
+
+To specify the application base directory path, you can use the ``appbase`` option. For example::
```

Comment:
`--appbase` is actually a complicated option to use. I'd avoid documenting it here. There is a different argument, `--project` (short form `-p`) which can be used to run commands out of a different project, for example:

```
dnx -p tests\MyTestProject test
```